### PR TITLE
State Modelling

### PR DESCRIPTION
**Pull request by ashfurrow**
Very early right now, but wanted to get something in to show intentions. 

TODO:

- [x] Set up initial view only once
- [x] Parameterize socket URL in labs settings
- [x] Fix tests
- [x] Add tests for new code

### CODE REVIEW

**Comment by ArtsyMobile** [comment]
<table>
  <thead>
    <tr>
      <th width="50"></th>
      <th width="100%">2 Warnings</th>
     </tr>
  </thead>
  <tbody>
        <tr>
      <td>:warning:</td>
      <td>PR is classed as Work in Progress</td>
    </tr>
        <tr>
      <td>:warning:</td>
      <td>Needs testing on a Phone if change is non-trivial</td>
    </tr>
    
  </tbody>
</table>
  

  


<p align="right" data-meta="generated_by_danger" data-base-commit="" data-head-commit="" >
  Generated by :no_entry_sign: <a href="https://github.com/danger/danger/">danger</a>
</p>

File: Artsy/Networking/Live_Auctions/LiveAuctionSocketCommunicator.swift [hunk 1]
```diff
@@ -65,8 +75,12 @@ private extension SocketSetup {
         socket.emit(.JoinSale, saleID)
 
         print("Listening for socket events.")
-        socket.on(.UpdateAuctionState) { data in
+        socket.on(.UpdateAuctionState) { [weak self] data in
             print("Updated auction state: \(data)")
+            // TODO: We need initial state when we join, this doesn't work until something on the server hapens :(
```
Comment (orta) [inline Artsy/Networking/Live_Auctions/LiveAuctionSocketCommunicator.swift]: I've heard that we may need to use the REST api for that

File: Artsy/View_Controllers/Live_Auctions/LiveAuctionViewController.swift [hunk 2]
```diff
@@ -87,7 +105,20 @@ class LiveAuctionViewController: UIViewController {
         progress.alignBottomEdgeWithView(view, predicate: "-165")
     }
 
-    func jumpToLiveLot(sender: AnyObject) {
+    func setupWithInitialData() {
+        // Make sure we only initialize with initial data once.
+        guard hasBeenSetup == false else { return }
+        defer { hasBeenSetup = true }
```
Comment (orta) [inline Artsy/View_Controllers/Live_Auctions/LiveAuctionViewController.swift]: nice

File: Artsy/View_Controllers/Live_Auctions/LiveAuctionViewController.swift [hunk 3]
```diff
@@ -111,26 +142,29 @@ class LiveAuctionSaleLotsDataSource : NSObject, UIPageViewControllerDataSource {
         let auctionVC =  LiveAuctionLotViewController()
         guard let viewModel = salesPerson.lotViewModelForIndex(index) else { return nil }
         auctionVC.lotViewModel.update(viewModel)
-        auctionVC.auctionViewModel.update(salesPerson.auctionViewModel)
+        // TODO: Figoure how to update the auctionVC's auctionViewModel signal in a better way.
```
Comment (orta) [inline Artsy/View_Controllers/Live_Auctions/LiveAuctionViewController.swift]: typo, but not critical

**Comment by orta** [comment]
:+1:

File: Artsy/Networking/Live_Auctions/LiveAuctionSocketCommunicator.swift (hunk 1 above)
Comment (alloy) [inline Artsy/Networking/Live_Auctions/LiveAuctionSocketCommunicator.swift]: https://github.com/artsy/prediction/blob/master/api/routes/state.js

Comment (ashfurrow) [inline Artsy/Networking/Live_Auctions/LiveAuctionSocketCommunicator.swift]: Perfect, thanks :cake: :bow:

**Pull request merged**